chalice/local.py

```diff
@@ -109,98 +128,251 @@ def create_lambda_event(self, method, path, headers, body=None):
+            )
+            return lambda_event, lambda_context
+        auth_result = authorizer(auth_event, lambda_context)
+        authed = self._check_can_invoke_view_function(
+            route_entry, auth_result)
+        if authed:
+            lambda_event = self._update_lambda_event(lambda_event, auth_result)
+        else:
+            raise NotAuthorizedError(
+                {'x-amzn-RequestId': lambda_context.aws_request_id,
+                 'x-amzn-ErrorType': 'UnauthorizedException'})
+        return lambda_event, lambda_context
+
+    def _check_can_invoke_view_function(self, route_entry, auth_result):
+        # type: (RouteEntry, ResponseType) -> bool
+        policy = auth_result.get('policyDocument', {})
+        statements = policy.get('Statement', [])
+        # Look for a statement that allows us to invoke the route_entry
+        # it will be in the form local:local/METHOD/uri the URI is always
+        # prefixed with / so we do not need to explicitly include it in our
+        # pattern.
+        # The method can be either an explicit method such as POST, GET, etc...
+        # or it can be a *. We also need to match for any prefix since the
+        # user can return any policy at all from the custom authorizer.
+        method_pattern = r'(?:\*|%s)' % route_entry.method
+        route_entry_pattern = r'^.*?/%s%s$' % (method_pattern,
+                                               route_entry.uri_pattern)
+        route_entry_re = re.compile(route_entry_pattern)
+        # Look through all statements for one that allows us to invoke this
+        # particular view function.
+        for statement in statements:
+            if statement.get('Effect') == 'Allow' and \
+               statement.get('Action') == 'execute-api:Invoke':
+                for resource in statement.get('Resource'):
+                    if route_entry_re.match(resource):
+                        return True
+        return False
+
+    def _authorizer_and_route_for_event(self, lambda_event):
+        # type: (EventType) -> Tuple[Optional[Any], Optional[RouteEntry]]
+        # Authorizer had to be made into an Any type since mypy couldn't
+        # detect that app.ChaliceAuthorizer was callable.
+        resource_path = lambda_event.get(
+            'requestContext', {}).get('resourcePath')
+        http_method = lambda_event['requestContext']['httpMethod']
+        try:
+            route_entry = self._app_object.routes[resource_path][http_method]
+        except KeyError:
+            # If there a key error is raised when trying to get the route entry
+            # then this route does not support this method. A method error
+            # will be raised by the chalice handler method. We can ignore it
+            # here by returning no authorizer to avoid duplicating the logic.
+            return None, None
+        authorizer = route_entry.authorizer
+        return authorizer, route_entry
```

Comment:
If `authorizer` is an attribute of `route_entry` why are you not just returning `route_entry`?